pandas/_typing.py

```diff
@@ -29,5 +29,11 @@
 Dtype = Union[str, np.dtype, ExtensionDtype]
 FilePathOrBuffer = Union[str, Path, IO[AnyStr]]
 
+# Type for index and columns of DataFrame
+Axes = Iterable[Union[ABCIndexClass, Iterable[str]]]
+
+# Type for data of DataFrame
+Data = TypeVar("Data", ABCDataFrame, dict, np.ndarray)
```

Comment:
Trying to add TypeVar for `data`. I know it's incomplete, some help is appreciated.
@WillAyd @jreback

Reply:
I would take this out of pandas._typing for now as it's a little too generic. The DataFrame constructor will have its own set of requirements so just localize it to that without TypeVar

Reply:
kk.